lib/types/protocol.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package common
+package types
 
 type Protocol struct {
```

Comment:
I don't see that sharing structs between the back end and front end is an issue.  If we were using net.IP as is in both the front end and back end I'm sure we wouldn't question that.